.github/workflows/server-redis.yml

```diff
@@ -0,0 +1,49 @@
+name: libs/server-sdk-redis-source
+
+on:
+  push:
+    branches: [ main ]
+    paths-ignore:
+      - '**.md' #Do not need to run CI for markdown changes.
+  pull_request:
+    branches: [ "main", "feat/**" ]
+    paths-ignore:
+      - '**.md'
+
+jobs:
+  build-test-redis:
+    runs-on: ubuntu-22.04
+    services:
+      redis:
+        image: redis
+        ports:
+          - 6379:6379
+    steps:
+      - uses: actions/checkout@v3
+      - uses: ./.github/actions/ci
+        with:
+          cmake_target: launchdarkly-cpp-server-redis-source
+          simulate_release: true
+  build-redis-mac:
+    runs-on: macos-12
+    steps:
+      - uses: actions/checkout@v3
+      - uses: ./.github/actions/ci
+        with:
+          cmake_target: launchdarkly-cpp-server-redis-source
+          platform_version: 12
```

Comment:
Unless there's a quickfix for running Redis on the macos and windows runners, I'm going to file a ticket for adding it in the future.